apps/server/src/shared/repo/task/task.repo.ts

```diff
@@ -183,10 +170,18 @@ export class TaskRepo extends BaseRepo<Task> {
 
 	private async findTasksAndCount(query: FilterQuery<Task>, options?: IFindOptions<Task>): Promise<Counted<Task[]>> {
 		const { pagination, order } = options || {};
+
+		// In order to solve pagination missmatches we apply a default order by _id. This is necessary
+		// because other fields like the dueDate can be equal or null.
+		// When pagination is used, sorting takes place on every page and if ambiguous leads to unwanted results.
+		// Note: Indexes for dueDate and for _id do exist but there's no combined index.
+		// This is okay, because the combined index would be too expensive for the particular purpose here.
+		const orderById = { _id: SortOrder.asc };
+
 		const [taskEntities, count] = await this._em.findAndCount(Task, query, {
 			offset: pagination?.skip,
 			limit: pagination?.limit,
-			orderBy: order as QueryOrderMap<Task>,
+			orderBy: { ...order, ...orderById },
```

Comment:
{...orderById, ...order } that we can override the _id sorting if we like or not?

Reply:
As I found out the order of the criteria does indeed matter. In order to make the sorting work properly we have to apply the dueDate order criterium first. If we want to make the id criterium overridable we have to implement that differently.

Reply:
I updated that.